test/e2e/apimachinery/custom_resource_definition.go

```diff
@@ -42,29 +51,143 @@ var _ = SIGDescribe("CustomResourceDefinition resources", func() {
+			defer func() {
+				err = fixtures.DeleteCustomResourceDefinition(crd, apiExtensionClient)
+				framework.ExpectNoError(err, "deleting CustomResourceDefinition")
+			}()
+
+			selectorListOpts := metav1.ListOptions{LabelSelector: "e2e-list-test-uuid=" + testUUID}
+			list, err := apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().List(selectorListOpts)
+			framework.ExpectNoError(err, "listing CustomResourceDefinitions")
+			framework.ExpectEqual(len(list.Items), testListSize)
+			for _, actual := range list.Items {
+				var expected *v1beta1.CustomResourceDefinition
+				for _, e := range crds {
+					if e.Name == actual.Name && e.Namespace == actual.Namespace {
+						expected = e
+					}
+				}
+				framework.ExpectNotEqual(expected, nil)
+				if !equality.Semantic.DeepEqual(actual.Spec, expected.Spec) {
+					e2elog.Failf("Expected CustomResourceDefinition in list with name %s to match crd created with same name, but got different specs:\n%s",
+						actual.Name, diff.ObjectReflectDiff(expected.Spec, actual.Spec))
 				}
+			}
+
+			// Use delete collection to remove the CRDs
+			err = fixtures.DeleteCustomResourceDefinitions(selectorListOpts, apiExtensionClient)
+			framework.ExpectNoError(err, "deleting CustomResourceDefinitions")
+			_, err = apiExtensionClient.ApiextensionsV1beta1().CustomResourceDefinitions().Get(crd.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err, "getting remaining CustomResourceDefinition")
+		})
+
+		/*
+			Release : v1.16
+			Testname: Custom Resource Definition, status sub-resource
+			Description: Create a API extension client, create a custom resource definition and then read, update and patch its status sub-resource. API server MUST be able to perform the operations against the status sub-resource.
+		*/
+		ginkgo.It("getting/updating/patching custom resource definition status sub-resource works ", func() {
+			config, err := framework.LoadConfig()
+			framework.ExpectNoError(err, "loading config")
+			apiExtensionClient, err := clientset.NewForConfig(config)
+			framework.ExpectNoError(err, "initializing apiExtensionClient")
+			dynamicClient, err := dynamic.NewForConfig(config)
+			framework.ExpectNoError(err, "initializing dynamic client")
+
+			// Create CRD and waits for the resource to be recognized and available.
+			crd := fixtures.NewRandomNameCustomResourceDefinition(v1beta1.ClusterScoped)
+			crd, err = fixtures.CreateNewCustomResourceDefinition(crd, apiExtensionClient, f.DynamicClient)
+			framework.ExpectNoError(err, "creating CustomResourceDefinition")
+			defer func() {
+				err = fixtures.DeleteCustomResourceDefinition(crd, apiExtensionClient)
+				framework.ExpectNoError(err, "deleting CustomResourceDefinition")
 			}()
+
+			// Use dynamic client to read the status sub-resource since typed client does not expose get.
+			gvr := schema.GroupVersionResource{Group: v1beta1.GroupName, Version: "v1beta1", Resource: "customresourcedefinitions"}
```

Comment:
`gvr := v1beta1.SchemeGroupVersion.WithResource("customresourcedefinitions")`? that will help us notice when switching this to test v1